src/main/java/org/broadinstitute/hellbender/dev/utils/ReadRecord.java

```diff
@@ -0,0 +1,274 @@
+        if (null != alignmentStart) {
+          // convert from 0-based to 1-based
+          this.setAlignmentStart(alignmentStart.intValue()+1);
+        }
+        Boolean reverseStrand = position.getReverseStrand();
+        if (null != reverseStrand) {
+          this.setReadNegativeStrandFlag(reverseStrand);
+        }
+      }
+      Integer mappingQuality = alignment.getMappingQuality();
+      if (null != mappingQuality) {
+        this.setMappingQuality(mappingQuality);
+      }
+      List<CigarUnit> cigar = alignment.getCigar();
+      if (null != cigar) {
+        this.setCigar(
+            new htsjdk.samtools.Cigar(read.getAlignment().getCigar().stream().map(ReadRecord::genToSam).collect(Collectors.toList())));
+      }
+    }
+    Boolean secondaryAlignment = read.getSecondaryAlignment();
+    if (null != secondaryAlignment) {
+      this.setNotPrimaryAlignmentFlag(secondaryAlignment);
+    }
+    Boolean supplementaryAlignment = read.getSupplementaryAlignment();
+    if (null != supplementaryAlignment) {
+      this.setSupplementaryAlignmentFlag(supplementaryAlignment);
+    }
+    String alignedSequence = read.getAlignedSequence();
+    if (null != alignedSequence) {
+      this.setReadString(alignedSequence);
+    }
+    List<Integer> alignedQuality = read.getAlignedQuality();
+    if (null != alignedQuality) {
+      this.setBaseQualities(integerListToByteArray(alignedQuality));
+    }
+    if (read.getNumberReads()>1) {
+      // 0x1: template having multiple segments in sequencing
+      this.setReadPairedFlag(true);
+      // 0x80:  the last segment in the template
+      this.setSecondOfPairFlag(read.getReadNumber() + 1 == read.getNumberReads());
+      // 0x8: next segment in the template unmapped
+      this.setMateUnmappedFlag(read.getNextMatePosition()==null);
+    }
+    Position nextMatePosition = read.getNextMatePosition();
+    if (null != nextMatePosition) {
+      String referenceName = nextMatePosition.getReferenceName();
+      if (null != referenceName) {
+        this.setMateReferenceName(referenceName);
+      }
+      Long position = nextMatePosition.getPosition();
+      if (null != position) {
+        // convert from 0-based to 1-based
+        this.setMateAlignmentStart(position.intValue()+1);
+      }
+      Boolean reverseStrand = nextMatePosition.getReverseStrand();
+      if (null != reverseStrand) {
+        this.setMateNegativeStrandFlag(reverseStrand);
+      }
+    }
+    Map<String, List<String>> info = read.getInfo();
+    if (null != info) {
+      for (Map.Entry<String, List<String>> entry : info.entrySet()) {
+        String key = entry.getKey();
+        List<String> value = entry.getValue();
+        try {
+          Tag tag = Tag.valueOf(key);
+          this.setAttribute(
+              key, null == tag ? Iterables.getOnlyElement(value) : tag.convert(value));
+        } catch (IllegalArgumentException x) {
+          // X,Y, or Z tag. Not sure what to do.
```

Comment:
We'll need to have a discussion around that; Reads and SAMRecords don't match 1:1 so not all the information can cross over. It can be useful to take an object and convert it even if we can't carry everything over - especially for things like the XYZ tags, which aren't standardized.